### PR TITLE
fix(style): Use better text selection color

### LINKED ISSUE #2435: Some dark mode highlighted code is invisible
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:not

### What specific section or headline is this issue about?

The code examples section

### What information was incorrect, unhelpful, or incomplete?

When viewing the page dark mode, the color used for the selected text that is within a parenthetical is essentially invisible. See this example:

highlighted code text with some words invisible

And selecting the text as in the next image shows the missing words from the previous slide.

the code selected in order to reveal the missing words

### What did you expect to see?

I was hoping to see highlighted code (text) with no colors any darker than a contrast value of $Lc \pm50$ ( $Lc$ is APCA lightness contrast)

### Do you have any supporting links, references, or citations?

Human vision system's contrast perception is largely driven by special frequency in other words, smaller and thinner stimuli result in much lower contrast sensitivity. Obviously, this applies very much to small and thin text.

Secondly, WCAG 2.x contrast math is not capable of calculating for dark mode, and it's not even close. WCAG 2.x contrast can be off by as much as 250% for dark colors.

If you want or need an automated way to determine contrast for dark colors, use an [APCA based contrast Calculator](https://www.myndex.com/APCA/). 

In the case of code, it's important things like periods and semicolons that have very small details differentiating them from similar elements: ` ., and ;: ` period verses comma or ; verses : need more luminance contrast to differentiate, than ` A vs W `

In my personal development environment, my background is black ` #000000 ` and punctuation is white, usually ` #f4f4f4 `  to ` #ffffff `. APCA dark mode calls this $Lc\ −100$ to $Lc\ -108$ ($-108$ is max -- note that APCA reports dark mode colors with a minus sign to distinguish them from light mode colors). 

The darkest color I use anywhere is for comments, and that is Lc&nbsp;-55 and for me and my vision and the size of code letters, I really can't deal with anything darker than $Lc\ -50$, and I would suggest that no color in a color palette for code highlighting ever be less than $Lc\ -50$, but if it's things that you need to see should be much more than $Lc \pm60$. $Lc \pm75$ to $Lc \pm90$ is fairly ideal for small text.

### Do you have anything more you want to share?

You might also like this article, a comprehensive overview of the important aspects of text color and contrast:  https://www.smashingmagazine.com/2022/09/realities-myths-contrast-color/

And the very short introduction to APCA contrast is [**_"Why APCA?"_**](https://git.apcacontrast.com/documentation/WhyAPCA)

Thank you for reading

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_colon_not`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/:not
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_colon_not/index.md
* Last commit: https://github.com/mdn/content/commit/4e002d26cb032c915aeee366f922f23cbacd8bf1
* Document last modified: 2023-02-11T03:07:14.000Z

</details>

### PR DESCRIPTION
**Pull request by NiedziolkaMichal**
This PR adds more suitable text selection colors in code mirror editors.

**Before:**
image
image

**After:**
image
image

[Myndex](https://github.com/Myndex) Could you please take a look, if those colors look fine?
Fixes [#2435](https://github.com/mdn/interactive-examples/issues/2435)

### CODE REVIEW

**Comment by teoli2003** [comment]
Ping @myndex. I'm ok to merge this if there is no more feedback.

**Review by teoli2003 — APPROVED** [review]
On second thought, this is much much better than before. Even if we could do better, we can do it in a follow-up. No need to hold this up for perfection.

**Comment by teoli2003** [comment]
Pinging: @wbamberg and @schalkneethling to get this merged.

File: editor/css/editor-libs/codemirror-override.css
Comment (wbamberg) [inline editor/css/editor-libs/codemirror-override.css]:
```suggestion
  background-color: var(--selection-background-color) !important;
```

...and the corresponding changes where this is referenced

**Comment by wbamberg** [comment]
> Pinging: @wbamberg and @schalkneethling to get this merged.

I think this seems like a good change, but I'm not a maintainer for mdn/bob.

File: editor/css/editor-libs/codemirror-override.css
Comment (NiedziolkaMichal) [reply editor/css/editor-libs/codemirror-override.css]: Thank you, it's fixed now.

**Comment by Myndex** [comment]
Hi @teoli2003 sorry for the delay in response,

Generally looks much better! and works...

However a slight tweak could help some color vision deficiencies just a little bit better. In dark mode the variable color `--code-token-attribute-name` is set to `#ff97a0`, and for individuals with protanopia it ends up being darker and indistinguishable from comments.

A very minor tweak to `#ffbb88` makes a big difference for protanopia.    
On the left is the current color `#ff97a0` for attribute name, on the right is `#ffbb88`, the modified version. 

read text on the left is CURRENT, on the right little bit more orange to make it more visible for protanopia


In the image below, processed to simulate various forms of color vision deficiency using the clinically-accurate, peer-reviewed Brettel model. Upper left is standard vision lower right is protanopia. (look at the words `String` and `.log`):

processed to simulate protanopia and other color vision deficiencies, looking at the pro nope here on the lower right we see that the keywords string and dialogue are much darker than the current sample but have better situation in our brighter when we've added a little green

**Comment by NiedziolkaMichal** [comment]
@Myndex I have applied the requested change

**Review by queengooborg — APPROVED** [review]

**Pull request merged**
